### PR TITLE
Adapt seeding cpu test to acts v37.3.0

### PR DESCRIPTION
**Pull request by CarloVarni**
This adapts the seeding cpu test and the atlas cuts to Acts v37.3.0. since the ACTS seeding has changed quite a lot, I've also added some comments to explain what we do.

From local test, the seed matching between Acts and traccc is ~0.9995

@krasznaa

### CODE REVIEW

**Comment by krasznaa** [comment]
image

No way! That was time very well spent with you this afternoon... 😄

File: tests/cpu/compare_with_acts_seeding.cpp [hunk 1]
```diff
@@ -42,6 +41,57 @@
...
+    std::size_t size_impl() const { return storage().size(); }
+    float x_impl(std::size_t idx) const { return storage()[idx]->x(); }
+    float y_impl(std::size_t idx) const { return storage()[idx]->y(); }
+    float z_impl(std::size_t idx) const { return storage()[idx]->z(); }
+    float varianceR_impl(std::size_t idx) const {
+        return storage()[idx]->varianceR;
+    }
+    float varianceZ_impl(std::size_t idx) const {
+        return storage()[idx]->varianceZ;
+    }
```
Comment (krasznaa) [inline tests/cpu/compare_with_acts_seeding.cpp:70]: Interesting... 🤔 I **think** we should also be able to skip defining a `SpacePoint` class entirely, and just use `traccc::spacepoint` directly, no?

In any case, that's something that I'll be pretty interested to try for myself. As the spacepoints will soon need to go SoA as well anyway... 🤔

**Review by krasznaa — APPROVED** [review]
I'm happy with it myself. 😄 But let's wait for @beomki-yeo to have a look as well.

File: tests/cpu/compare_with_acts_seeding.cpp (hunk 1 above)
Comment (CarloVarni) [reply tests/cpu/compare_with_acts_seeding.cpp:70]: indeed you can do it

**Review by beomki-yeo — APPROVED** [review]
What an amazing work.. Thanks!

**Pull request merged**